p2p/net/swarm/swarm_metrics.go

```diff
@@ -108,12 +108,28 @@ func appendConnectionState(tags []string, cs network.ConnectionState) []string {
 	return tags
 }
 
-func (m *metricsTracer) OpenedConnection(dir network.Direction, p crypto.PubKey, cs network.ConnectionState) {
+func getIPVersion(addr ma.Multiaddr) string {
+	var version string
+	ma.ForEach(addr, func(c ma.Component) bool {
+		if c.Protocol().Code == ma.P_IP4 {
+			version = "ip4"
+			return false
+		} else if c.Protocol().Code == ma.P_IP6 {
+			version = "ip6"
+			return false
+		}
+		return true
+	})
```

Comment:
@marten-seemann ```addr.ValueForProtocol(ma.P_IP4)``` allocates on heap, so doing it this way.

Reply:
Yes that's fine. Our multiaddr API isn't the cleanest...